lib/amqp/gen/consumer.ex

```diff
@@ -242,4 +259,21 @@ defmodule Amqpx.Gen.Consumer do
 
       state
   end
+
+  @spec draining? :: boolean()
+  def draining?, do: signal_handler().draining?
+
+  def stopping?, do: signal_handler().stopping?
+
+  defp cancel?(%{cancel?: true} = state, _), do: state
+
+  defp cancel?(state, consumer_tag) do
+    Logger.info("Cancelling consumer #{consumer_tag}")
+
+    if draining?() do
+      Basic.cancel(state.channel, consumer_tag)
+    end
+
+    %{state | cancel?: true}
```

Comment:
If `cancel?` is to mark that it's cancelling then I would call this `cancelling?` to match the other vars, otherwise I misunderstood the logic